enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -308,14 +312,61 @@ def _create_kernel_namespace(self, service_account_name: str) -> str:
                     body = client.V1DeleteOptions(
                         grace_period_seconds=0, propagation_policy="Background"
                     )
-                    client.CoreV1Api().delete_namespace(name=namespace, body=body)
+                    client.CoreV1Api(api_client=kubernetes_client).delete_namespace(name=namespace, body=body)
                     self.log.warning(f"Deleted kernel namespace: {namespace}")
                 else:
                     reason = f"Error occurred creating namespace '{namespace}': {err}"
                 self.log_and_raise(http_status_code=500, reason=reason)
 
         return namespace
 
+    def _create_service_account_if_not_exists(self, namespace: str, service_account_name: str) -> None:
+        service_account_list_in_namespace: client.V1ServiceAccountList = client\
+            .CoreV1Api(api_client=kubernetes_client)\
+            .list_namespaced_service_account(namespace=namespace)
+
+        service_accounts_in_namespace: List[client.V1ServiceAccount] = service_account_list_in_namespace.items
+        service_account_names_in_namespace: List[str] = list(map(lambda svcaccount: svcaccount.metadata.name, service_accounts_in_namespace))
+
+        if service_account_name not in service_account_names_in_namespace:
+            service_account_metadata = {
+                "name": service_account_name
+            }
+            service_account_to_create: client.V1ServiceAccount = client.V1ServiceAccount(
+                kind="ServiceAccount",
+                metadata=service_account_metadata
+            )
+
+            client\
+                .CoreV1Api(api_client=kubernetes_client)\
+                .create_namespaced_service_account(namespace=namespace, body=service_account_to_create)
+
+            self.log.info(f"Created service account {service_account_name} in namespace {namespace}")
+
+    def _forward_role_to_remote(self) -> None:
```

Comment:
A few thoughts:

1. I don't think we (Lacework) want to grant a ClusterRole to JEG. At most, I think we want to grant roles to a handful of namespaces. 
2. I don't think we should be copying a role from one place to another, this seems non-transparent.
3. I would recommend adding a static rolebinding to a specific namespace.
3. I would recommend adding an option to _not_ create the service account, allowing the user to supply their own service account. This lets people customize as they choose.